### PR TITLE
Move projects that target netcoreapp2.1 to netcoreapp3.1

### LINKED ISSUE #11502: Move NuGet test projects from netcoreapp2.1 to netcoreap3.1
netcoreapp2.1 is deprecated, so we can move to netcoreapp3.1

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug

Fixes: https://github.com/NuGet/Home/issues/11502

Regression? Last working version:

## Description

netcoreapp2.1 is EOL, we can move to netcoreapp3.1 given that we're using this framework only in test projects and the runnables (meaning, not libraries). 

There's an issue that has to do with minimizing the frameworks we're targetting: https://github.com/NuGet/Home/issues/9886, which can lead to speeding up our build. 

Here's the current state of the frameworks: 

```txt
  NuGet.CommandLine.FuncTest.csproj = net472
  NuGet.MSSigning.Extensions.FuncTest.csproj = net472
  NuGet.CommandLine.Test.csproj = net472
  NuGet.Indexing.Test.csproj = net472
  NuGet.MSSigning.Extensions.Test.csproj = net472
  NuGet.PackageManagement.UI.Test.csproj = net472
  NuGet.PackageManagement.VisualStudio.Test.csproj = net472
  NuGet.SolutionRestoreManager.Test.csproj = net472
  NuGet.Tools.Test.csproj = net472
  NuGet.VisualStudio.Common.Test.csproj = net472
  NuGet.VisualStudio.Implementation.Test.csproj = net472
  NuGet.VisualStudio.Internal.Contracts.Test.csproj = net472
  NuGet.VisualStudio.Test.csproj = net472
  NuGetConsole.Host.PowerShell.Test.csproj = net472
  Dotnet.Integration.Test.csproj = netcoreapp5.0
  Msbuild.Integration.Test.csproj = net472
  NuGet.Commands.FuncTest.csproj = net472;netcoreapp3.1;netcoreapp5.0
  NuGet.Packaging.FuncTest.csproj = net472;netcoreapp3.1;netcoreapp5.0
  NuGet.Protocol.FuncTest.csproj = net472;netcoreapp3.1;netcoreapp5.0
  NuGet.Signing.CrossFramework.Test.csproj = net472;netcoreapp5.0
  NuGet.XPlat.FuncTest.csproj = net472;netcoreapp3.1;netcoreapp5.0
  Microsoft.Build.NuGetSdkResolver.Test.csproj = net472;netcoreapp3.1;netcoreapp5.0
  NuGet.Build.Tasks.Console.Test.csproj = net472;netcoreapp3.1;netcoreapp5.0
  NuGet.Build.Tasks.Pack.Test.csproj = net472;netcoreapp3.1;netcoreapp5.0
  NuGet.Build.Tasks.Test.csproj = net472;netcoreapp3.1;netcoreapp5.0
  NuGet.CommandLine.Xplat.Tests.csproj = net472;netcoreapp3.1;netcoreapp5.0
  NuGet.Commands.Test.csproj = net472;netcoreapp3.1;netcoreapp5.0
  NuGet.Common.Test.csproj = net472;netcoreapp3.1
  NuGet.Configuration.Test.csproj = net472;netcoreapp3.1
  NuGet.Credentials.Test.csproj = net472
  NuGet.DependencyResolver.Core.Tests.csproj = net472;netcoreapp3.1;netcoreapp5.0
  NuGet.Frameworks.Test.csproj = net472;netcoreapp3.1
  NuGet.LibraryModel.Tests.csproj = net472;netcoreapp3.1
  NuGet.PackageManagement.Test.csproj = net472;netcoreapp3.1
  NuGet.Packaging.Test.csproj = net472;netcoreapp3.1;netcoreapp5.0
  NuGet.ProjectModel.Test.csproj = net472;netcoreapp3.1;netcoreapp5.0
  NuGet.Protocol.Tests.csproj = net472;netcoreapp3.1;netcoreapp5.0
  NuGet.Resolver.Test.csproj = net472;netcoreapp3.1;netcoreapp5.0
  NuGet.Shared.Tests.csproj = net472;netcoreapp3.1
  NuGet.Versioning.Test.csproj = net472;netcoreapp3.1
  API.Test.csproj = net472
  GenerateLicenseList.csproj = net5.0
  GenerateTestPackages.csproj = net472
  NuGet.StaFact.csproj = net472
  SampleCommandLineExtensions.csproj = net472
  TestableCredentialProvider.csproj = net472
  TestablePlugin.csproj = net472;netcoreapp3.1;netcoreapp5.0
  TestableVSCredentialProvider.csproj =
  Test.Utility.csproj = net472;netcoreapp3.1;netcoreapp5.0
  NuGet.CommandLine.csproj = net472
  NuGet.Console.csproj =
  NuGet.Indexing.csproj = net472
  NuGet.MSSigning.Extensions.csproj = net472
  NuGet.PackageManagement.PowerShellCmdlets.csproj =
  NuGet.PackageManagement.UI.csproj =
  NuGet.PackageManagement.VisualStudio.csproj =
  NuGet.SolutionRestoreManager.csproj =
  NuGet.Tools.csproj =
  NuGet.VisualStudio.Client.csproj =
  NuGet.VisualStudio.Common.csproj = net472
  NuGet.VisualStudio.Contracts.csproj = netstandard2.0
  NuGet.VisualStudio.Implementation.csproj = net472
  NuGet.VisualStudio.Internal.Contracts.csproj = net472
  NuGet.VisualStudio.Interop.csproj = net472
  NuGet.VisualStudio.OnlineEnvironment.Client.csproj =
  NuGet.VisualStudio.csproj = net472
  NuGetConsole.Host.PowerShell.csproj =
  Microsoft.Build.NuGetSdkResolver.csproj = net472;netstandard2.0;netcoreapp5.0
  NuGet.Build.Tasks.Console.csproj = net472;netcoreapp3.1;netcoreapp5.0
  NuGet.Build.Tasks.Pack.csproj = net472;netstandard2.0
  NuGet.Build.Tasks.csproj = net472;netstandard2.0;netcoreapp5.0
  NuGet.CommandLine.XPlat.csproj = net472;netcoreapp3.1;netcoreapp5.0
  NuGet.Commands.csproj = net472;netstandard2.0;netcoreapp5.0
  NuGet.Common.csproj = net472;netstandard2.0;net45
  NuGet.Configuration.csproj = net472;netstandard2.0;net45
  NuGet.Credentials.csproj = net472;netstandard2.0;netcoreapp5.0
  NuGet.DependencyResolver.Core.csproj = net472;netstandard2.0;netcoreapp5.0
  NuGet.Frameworks.csproj = net472;netstandard2.0;net40
  NuGet.LibraryModel.csproj = net472;netstandard2.0
  NuGet.Localization.csproj = netstandard2.0
  NuGet.PackageManagement.csproj = net472;netstandard2.0
  NuGet.Packaging.Core.csproj = net472;netstandard2.0;netcoreapp5.0
  NuGet.Packaging.Extraction.csproj = net45
  NuGet.Packaging.csproj = net472;netstandard2.0;netcoreapp5.0
  NuGet.ProjectModel.csproj = net472;netstandard2.0;netcoreapp5.0
  NuGet.Protocol.csproj = net472;netstandard2.0;netcoreapp5.0
  NuGet.Resolver.csproj = net472;netstandard2.0;netcoreapp5.0
  NuGet.Versioning.csproj = net472;netstandard2.0;net45
  NuGet.Console.TestContract.csproj = net472
  NuGet.OptProf.csproj = net472
  NuGet.PackageManagement.UI.TestContract.csproj = net472
  NuGet.Tests.Apex.csproj = net472
```

src projects: 
```
  NuGet.CommandLine.csproj = net472
  NuGet.Console.csproj =
  NuGet.Indexing.csproj = net472
  NuGet.MSSigning.Extensions.csproj = net472
  NuGet.PackageManagement.PowerShellCmdlets.csproj =
  NuGet.PackageManagement.UI.csproj =
  NuGet.PackageManagement.VisualStudio.csproj =
  NuGet.SolutionRestoreManager.csproj =
  NuGet.Tools.csproj =
  NuGet.VisualStudio.Client.csproj =
  NuGet.VisualStudio.Common.csproj = net472
  NuGet.VisualStudio.Contracts.csproj = netstandard2.0
  NuGet.VisualStudio.Implementation.csproj = net472
  NuGet.VisualStudio.Internal.Contracts.csproj = net472
  NuGet.VisualStudio.Interop.csproj = net472
  NuGet.VisualStudio.OnlineEnvironment.Client.csproj =
  NuGet.VisualStudio.csproj = net472
  NuGetConsole.Host.PowerShell.csproj =
  Microsoft.Build.NuGetSdkResolver.csproj = net472;netstandard2.0;netcoreapp5.0
  NuGet.Build.Tasks.Console.csproj = net472;netcoreapp3.1;netcoreapp5.0
  NuGet.Build.Tasks.Pack.csproj = net472;netstandard2.0
  NuGet.Build.Tasks.csproj = net472;netstandard2.0;netcoreapp5.0
  NuGet.CommandLine.XPlat.csproj = net472;netcoreapp3.1;netcoreapp5.0
  NuGet.Commands.csproj = net472;netstandard2.0;netcoreapp5.0
  NuGet.Common.csproj = net472;netstandard2.0;net45
  NuGet.Configuration.csproj = net472;netstandard2.0;net45
  NuGet.Credentials.csproj = net472;netstandard2.0;netcoreapp5.0
  NuGet.DependencyResolver.Core.csproj = net472;netstandard2.0;netcoreapp5.0
  NuGet.Frameworks.csproj = net472;netstandard2.0;net40
  NuGet.LibraryModel.csproj = net472;netstandard2.0
  NuGet.Localization.csproj = netstandard2.0
  NuGet.PackageManagement.csproj = net472;netstandard2.0
  NuGet.Packaging.Core.csproj = net472;netstandard2.0;netcoreapp5.0
  NuGet.Packaging.Extraction.csproj = net45
  NuGet.Packaging.csproj = net472;netstandard2.0;netcoreapp5.0
  NuGet.ProjectModel.csproj = net472;netstandard2.0;netcoreapp5.0
  NuGet.Protocol.csproj = net472;netstandard2.0;netcoreapp5.0
  NuGet.Resolver.csproj = net472;netstandard2.0;netcoreapp5.0
  NuGet.Versioning.csproj = net472;netstandard2.0;net45
  ```
  
## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [x] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [ ] N/A

### CODE REVIEW

**Review by erdembayar — APPROVED** [review]

File: test/NuGet.Core.FuncTests/NuGet.Protocol.FuncTest/HttpRetryHandlerTests.cs [hunk 1]
```diff
@@ -122,7 +115,7 @@ public async Task HttpRetryHandler_HandlesInvalidProtocol()
             var exception = await ThrowsException<HttpRequestException>(server);
 #if IS_CORECLR
             Assert.Null(exception.InnerException);
-#if (NETCORE3_0 || NETCORE5_0)
+#if (NETCOREAPP3_1 || NETCORE5_0)
```
Comment (zivkan) [inline test/NuGet.Core.FuncTests/NuGet.Protocol.FuncTest/HttpRetryHandlerTests.cs]:
```suggestion
#if NETCOREAPP3_1_OR_GREATER
```

**Review by zivkan — APPROVED** [review]

File: test/NuGet.Core.FuncTests/NuGet.Protocol.FuncTest/HttpRetryHandlerTests.cs (hunk 1 above)
Comment (nkolev92) [reply test/NuGet.Core.FuncTests/NuGet.Protocol.FuncTest/HttpRetryHandlerTests.cs]: Oh it's `GREATER`. 

I tried so many times with `OR_HIGHER`, but it didn't work and I wasn't gonna look at a bin log too much.

I'll fix it.

Comment (nkolev92) [reply test/NuGet.Core.FuncTests/NuGet.Protocol.FuncTest/HttpRetryHandlerTests.cs]: Apparently GH likes to duplicate comments?

Should I add a couple more for good measure?

File: test/NuGet.Core.Tests/NuGet.Common.Test/DateTimeUtilityTests.cs [hunk 2]
```diff
@@ -16,7 +16,6 @@ public static IEnumerable<object[]> GetData()
             {
                 new object[] { "1.23", TimeSpan.FromSeconds(1.23d) },
                 new object[] { "0", TimeSpan.FromMilliseconds(0.01d) }, // round down
-                new object[] { "1", TimeSpan.FromMilliseconds(0.56d) }, // round up
                 new object[] { "1", TimeSpan.FromMilliseconds(1.21d) }, // round down
```
Comment (dominoFire) [inline test/NuGet.Core.Tests/NuGet.Common.Test/DateTimeUtilityTests.cs:19]: How the corner is affected by netcoreapp3.1 upgrade? Is it just an Assert error ?

Comment (nkolev92) [reply test/NuGet.Core.Tests/NuGet.Common.Test/DateTimeUtilityTests.cs:19]: There's a rounding error I can't repro locally. 

It's a known consequence of https://github.com/dotnet/runtime/issues/13230, so I've been trying to figure out which test case it is.

Comment (nkolev92) [reply test/NuGet.Core.Tests/NuGet.Common.Test/DateTimeUtilityTests.cs:19]: I'll need to dig further, I was hoping I was guessing it right, but think we'll need to do some more work.

Comment (nkolev92) [reply test/NuGet.Core.Tests/NuGet.Common.Test/DateTimeUtilityTests.cs:19]: Fixed the test case to account for the different behavior on different frameworks.

**Review by jeffkl — APPROVED** [review]

**Comment by nkolev92** [comment]
Linux tests are timing out unfortunately. 

Unclear the problem is. I might need some tools about detetcting long running, but I won't have the bandwidth until QW so gonna pause on this until then. 

I don't expect a re-review to be necessary here.

**Comment by nkolev92** [comment]
The linux tests pass...but the script returns 1. 

So that's exciting. I need to dig more here.

**Pull request merged**